timemachine/fe/single_topology.py

```diff
@@ -1018,27 +1171,83 @@ def assert_chiral_consistency_and_validity(
         assert canonicalize_bond((c, j)) in dst_bond_idxs
         assert canonicalize_bond((c, k)) in dst_bond_idxs
 
-    src_chiral_restr_idx_set = ChiralRestrIdxSet(src_chiral_idxs)
-    dst_chiral_restr_idx_set = ChiralRestrIdxSet(dst_chiral_idxs)
-
     # ensure that we don't have any chiral inversions between src and dst end states
     assert len(src_chiral_restr_idx_set.allowed_set.intersection(dst_chiral_restr_idx_set.disallowed_set)) == 0
     assert len(dst_chiral_restr_idx_set.allowed_set.intersection(src_chiral_restr_idx_set.disallowed_set)) == 0
 
+
+def check_chiral_validity_impl(
+    atom_map: AtomMapMixin,
+    src_chiral_restr_idx_set: ChiralRestrIdxSet,
+    dst_chiral_restr_idx_set: ChiralRestrIdxSet,
+    src_bond_idxs: NDArray,
+    dst_bond_idxs: NDArray,
+):
+    """Assert that we can directly turn on the chiral volumes (after bonds) without staggering angles."""
+
     chiral_centers_in_mol_a = chiral_utils.find_chiral_atoms(atom_map.mol_a)
     chiral_centers_in_mol_b = chiral_utils.find_chiral_atoms(atom_map.mol_b)
 
     src_chiral_centers_in_mol_c = [atom_map.a_to_c[x] for x in chiral_centers_in_mol_a]
     dst_chiral_centers_in_mol_c = [atom_map.b_to_c[x] for x in chiral_centers_in_mol_b]
 
-    check_chiral_validity(src_chiral_centers_in_mol_c, dst_chiral_restr_idx_set, src_bond_idxs)
-    check_chiral_validity(dst_chiral_centers_in_mol_c, src_chiral_restr_idx_set, dst_bond_idxs)
+    check_chiral_validity_src_dst(src_chiral_centers_in_mol_c, dst_chiral_restr_idx_set, src_bond_idxs)
+    check_chiral_validity_src_dst(dst_chiral_centers_in_mol_c, src_chiral_restr_idx_set, dst_bond_idxs)
+
+
+def assert_chiral_consistency_and_validity(
+    atom_map: AtomMapMixin,
+    src_chiral_idxs: NDArray,
+    dst_chiral_idxs: NDArray,
+    src_bond_idxs: NDArray,
+    dst_bond_idxs: NDArray,
+):
+    """
+    Assert that the given the two end states chiral and bond idxs it would be both consistent and valid.
+
+    consistency: if there are no inversions at the end-states between chiral atoms and bonds are present
+    validity: if we can directly turn on the chiral volumes (after bonds) without staggering angles
+    """
 
+    src_chiral_restr_idx_set = ChiralRestrIdxSet(src_chiral_idxs)
+    dst_chiral_restr_idx_set = ChiralRestrIdxSet(dst_chiral_idxs)
 
-def verify_chiral_consistency_of_core(mol_a: Chem.Mol, mol_b: Chem.Mol, core: NDArray, forcefield):
+    assert_chiral_consistency(src_chiral_restr_idx_set, dst_chiral_restr_idx_set, src_bond_idxs, dst_bond_idxs)
+    check_chiral_validity_impl(
+        atom_map, src_chiral_restr_idx_set, dst_chiral_restr_idx_set, src_bond_idxs, dst_bond_idxs
+    )
+
+
+def verify_chiral_validity_of_core(mol_a: Chem.Mol, mol_b: Chem.Mol, core: NDArray, forcefield):
```

Comment:
Note breaking rename of this function (cc @badisa). This was done to standardize the definitions of "chiral consistency" and "chiral validity".

Reply:
Also note that this can additionally raise a `DummyGroupAssignmentError`, if there were no cores found that are compatible with chirally-valid dummy group assignments, using the placeholder forcefield.

It can still raise `ChiralConversionError`, but this should not happen typically (only if there's a case where the placeholder forcefield and MD forcefield yield different chiral atoms).